resources/assert_lib.brs

```diff
@@ -60,3 +61,96 @@ function __formatError(errMessage)
         stack: {}
     }
 end function
+
+sub __deepEquals(actual, expected, error)
+    if __roca_deepEquals(actual, expected) = true then
+        m.__pass()
+    else
+        m.__fail(m.formatError(error))
+    end if
+end sub
+
+'utility function to compare two objects/associative arrays with nested data structures
+function __roca_deepEquals(lhs as object, rhs as object) as boolean
+    ' because Brightsript does not support closures and we are doing recursion,
+    'we want to hold the function parameters in top-level state variables during each execution
+    updatedLeftState = invalid
+    updatedRightState = invalid
+    isEqual = true
+
+    ' base case
+    if type(lhs) <> type(rhs) then return false
```

Comment:
We may need some kind of `isNumeric` check in here as well, since floats can be directly compared to integers and we likely want `3.0` to be equal to `3`!